src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -649,23 +649,34 @@ public override void CloseAdditionalDocument(DocumentId documentId)
 
         public bool TryGetInfoBarData(DocumentId documentId, out IVsWindowFrame frame, out IVsInfoBarUIFactory factory)
         {
+            frame = null;
+            factory = null;
             if (documentId == null)
             {
-                frame = null;
-                factory = null;
-                return false;
+                var monitorSelectionService = ServiceProvider.GetService(typeof(SVsShellMonitorSelection)) as IVsMonitorSelection;
+                object value = null;
+                if (monitorSelectionService != null &&
+                   (ErrorHandler.Succeeded(monitorSelectionService.GetCurrentElementValue((uint)VSConstants.VSSELELEMID.SEID_DocumentFrame, out value)) ||
+                    ErrorHandler.Succeeded(monitorSelectionService.GetCurrentElementValue((uint)VSConstants.VSSELELEMID.SEID_WindowFrame, out value))))
```

Comment:
We really shouldn't use that ```documentId```. The scenario of a codefix/refactoring encountering an error when no documents are open is certainly the main driving scenario for this change, but if we're making this change in the no-document case then we should make the change consistently, for all cases. For example, if I have Main VS on Monitor 1 and Interactive on Monitor 2, and a codefix has a problem while I'm working in Interactive, I don't want to have to look over to Monitor 1 to see the infobar *sometimes* (iff there are no documents open over the entire VS process).

Also, does this work correctly in Peek? (I have no idea how Peek works.)